src/kbucket.rs

```diff
@@ -560,19 +560,19 @@ where
     /// Returns an iterator over the keys that are contained in a kbucket, specified by a log2 distance.
     pub fn nodes_by_distances(
         &mut self,
-        log2_distances: Vec<u64>,
+        log2_distances: &[u64],
         max_nodes: usize,
     ) -> Vec<EntryRefView<'_, TNodeId, TVal>> {
         let distances = log2_distances
-            .into_iter()
-            .filter(|&d| d > 0 && d <= (NUM_BUCKETS as u64))
+            .iter()
+            .filter(|&d| *d > 0 && *d <= (NUM_BUCKETS as u64))
```

Comment:
I think it might even be better to just user the iterator and not `Collect` it. 


Something like:
```rust
        let filtered_distances = log2_distances
            .iter()
            .filter(|&&d| d > 0 && d <= (NUM_BUCKETS as u64));
 ```
 Then instead of having two for loops, combine them into a single for loop over `filtered_distances`.

Reply:
That'll save an allocation, so I'm all for it :D